homeassistant/core.py

```diff
@@ -1457,6 +1469,8 @@ async def async_store(self) -> None:
             "unit_system": self.units.name,
```

Comment:
I can't comment on `async_load`, so commenting on closest line. 

Can we do a one time import of base url to external url?

In `async_load`, we can check `if "external_url" not in data`. If so, we know it was last saved in an older version of HA without external url (or else key exists and value is `None`). When the key is not there, register a listener for `EVENT_HOMEASSISTANT_START` in which we check if `http` integration is loaded, and update the config to set external url = base url.

Reply:
The problem is that we can't assume the base_url is external. Right now it depends a bit on the users set up if it contains an internal URL or an external URL. People have been playing/struggling with base URL because of NAT loopback issues.

Hence the current approach uses the current base URL as a fallback with quite a few tests.

https://github.com/home-assistant/core/blob/bf9200e951af5eeeb457b6d5ee57e14dbf203eea/homeassistant/helpers/network.py#L202-L228

Reply:
That said, I guess we can run some of these checks during import and decide which field to fill, either internal_url or external_url. I will take a look.